src/Core/Rendering/RenderController.cpp

```diff
@@ -708,12 +709,51 @@ namespace MxEngine
         blurInputOutput->Bind(textureId++);
         applyShader->SetUniform("inputTex", input->GetBoundId());
         applyShader->SetUniform("SSGITex", blurInputOutput->GetBoundId());
-
+         
         this->RenderToTexture(output, applyShader);
 
         std::swap(input, output);
     }
+    void RenderController::ApplyDepthOfFieldEffect(CameraUnit& camera, TextureHandle& inputOutput, TextureHandle& temporary0, TextureHandle& temporary1)
+    {
+        if (camera.Effects->GetFocusDistance() == 0.f)
+            return;
+        MAKE_SCOPE_PROFILER("RenderController::ApplyDepthOfFieldEffect()");
+
+        inputOutput->GenerateMipmaps();
```

Comment:
are you accessing mip levels?

Reply:
Nope but I was facing this error.  The error was gone after I used mipmap.
https://stackoverflow.com/questions/24562896/opengl-edges-of-textures-are-black